### PR TITLE
#187919519 A seller should be able to update order status

### PR DESCRIPTION
**Pull request by YvetteNyibuka**
## What does this PR do?
Update order status: Delivering or Canceling order

## Description of Task to be completed?
A seller should be able to update order/sales status

## How should this be manually tested?

- clone this rep
- navigate to the branch: ft-update-orderStatus-187919519
- install  dependences and have a .env file with necessary environment variables
- run command: npm run dev
- login as a seller and go to sales section of dashboard
- click on any sale and update its status

## What are the relevant pivotal tracker/Trello stories?
#187919519

### CODE REVIEW

File: src/pages/dashboard/seller/Sales/UpdateOrderModal.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+							onClick={() => dispatch(closeModal())}
+							className="px-4 py-2 text-sm bg-action-error rounded-md text-neutral-white"
+						>
+							Cancel
+						</button>
+						<button
+							type="submit"
+							className="px-4 py-2 text-sm text-neutral-white bg-action-success rounded-md"
+						>
+							Submit
```
Comment (GarrixA) [inline src/pages/dashboard/seller/Sales/UpdateOrderModal.tsx]: You can add loading state while submitting the form

**Review by GarrixA — CHANGES_REQUESTED** [review]

File: src/pages/dashboard/seller/Sales/UpdateOrderModal.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+						<label htmlFor="status" className="block text-sm font-medium">
+							Status
+						</label>
+						<select
+							id="status"
+							value={status}
+							onChange={(e) => dispatch(setStatus(e.target.value))}
+							className="mt-1 block w-full pl-3 pr-10 py-2 border-2 text-base mobile:text-sm rounded-md"
+						>
+							<option value="delivered">Delivered</option>
```
Comment (GarrixA) [inline src/pages/dashboard/seller/Sales/UpdateOrderModal.tsx:109]: Why only two statuses while we have three?

File: src/pages/dashboard/seller/Sales/UpdateOrderModal.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+					</div>
+					<div className="mb-2">
+						<label htmlFor="deliveryDate" className="block text-sm font-medium">
+							Delivery Date
+						</label>
+						<input
+							type="date"
+							id="deliveryDate"
+							value={deliveryDate.split('T')[0]}
+							onChange={(e) => dispatch(setDeliveryDate(e.target.value))}
```
Comment (GarrixA) [inline src/pages/dashboard/seller/Sales/UpdateOrderModal.tsx]: Better to use react hook forms

File: src/pages/dashboard/seller/Sales/UpdateOrderModal.tsx (hunk 2 above)
Comment (YvetteNyibuka) [reply src/pages/dashboard/seller/Sales/UpdateOrderModal.tsx:109]: By default, the status is pending right after the order is made but when we are updating the order what is only allowed is to cancel it or deliver it. That is how it works on backend.

File: src/redux/features/Sales/AllSaleSlice.ts [hunk 4]
```diff
@@ -34,30 +45,55 @@ export const getSales = createAsyncThunk(
...
+		try {
+			const { data } = await API.get(`/sales/${id}`);
+			return data;
+		} catch (error) {
+			return rejectWithValue(error);
+		}
+	},
+);
+
 type SalesResponse = {
```
Comment (Angemichel12) [inline src/redux/features/Sales/AllSaleSlice.ts]: It is good practice to put type in types directory.

File: test/pages/LandingProduct.test.tsx [hunk 5]
```diff
@@ -110,21 +109,4 @@ describe('Get all products', () => {
 		).toBeInTheDocument();
 		expect(screen.getByPlaceholderText('Maximum price')).toBeInTheDocument();
 	});
```
Comment (Angemichel12) [inline test/pages/LandingProduct.test.tsx]: Why Have you deleted this test?

File: test/pages/ProductsPage.test.tsx [hunk 6]
```diff
@@ -115,20 +114,4 @@ describe('Get all products', () => {
 		).toBeInTheDocument();
 		expect(screen.getByPlaceholderText('Maximum price')).toBeInTheDocument();
 	});
```
Comment (Angemichel12) [inline test/pages/ProductsPage.test.tsx]: Also Here.

File: src/pages/dashboard/seller/Sales/UpdateOrderModal.tsx (hunk 1 above)
Comment (YvetteNyibuka) [reply src/pages/dashboard/seller/Sales/UpdateOrderModal.tsx]: I have added one, when is pending we have caption of updating... but when not pending we have uodate.

File: test/pages/LandingProduct.test.tsx (hunk 5 above)
Comment (YvetteNyibuka) [reply test/pages/LandingProduct.test.tsx]: The test owner Bonheur told me to do so as it was bringing more issues

File: test/pages/ProductsPage.test.tsx (hunk 6 above)
Comment (YvetteNyibuka) [reply test/pages/ProductsPage.test.tsx]: The same here

File: src/pages/dashboard/seller/Sales/UpdateOrderModal.tsx (hunk 3 above)
Comment (YvetteNyibuka) [reply src/pages/dashboard/seller/Sales/UpdateOrderModal.tsx]: Resolved

File: src/redux/features/Sales/AllSaleSlice.ts (hunk 4 above)
Comment (YvetteNyibuka) [reply src/redux/features/Sales/AllSaleSlice.ts]: cool

File: src/pages/dashboard/seller/Sales/SalesPage.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+	const handlePageChange = (selectedPage: number) => {
+		setCurrentPage(selectedPage);
+	};
+
+	useEffect(() => {
+		dispatch(getSales())
+			.unwrap()
+			.catch((error) => {
+				console.log(error);
```
Comment (Angemichel12) [inline src/pages/dashboard/seller/Sales/SalesPage.tsx]: Remove this console.

File: src/pages/dashboard/seller/Sales/SalesPage.tsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+			});
+	}, [dispatch]);
+
+	const offset = currentPage * itemsPerPage;
+	const currentSales = sales.slice(offset, offset + itemsPerPage);
+
+	console.log('currentSales', currentSales);
```
Comment (Angemichel12) [inline src/pages/dashboard/seller/Sales/SalesPage.tsx]: And also this.

File: src/pages/dashboard/seller/Sales/SingleSale.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+		(state: RootState) => state.sales,
+	);
+
+	const { id } = useParams();
+	useEffect(() => {
+		if (id) {
+			dispatch(getSingleSale(id))
+				.unwrap()
+				.catch((error) => {
+					console.log(error);
```
Comment (Angemichel12) [inline src/pages/dashboard/seller/Sales/SingleSale.tsx]: remove this

File: src/pages/dashboard/seller/SellerProductsPage.tsx [hunk 10]
```diff
@@ -13,7 +13,9 @@ import { useAppDispatch, useAppSelector } from '../../../redux/hooks/hooks';
 
 const SellerProductsPage = () => {
 	const { isLoading } = useAppSelector((state) => state.product);
+	// eslint-disable-next-line react-hooks/exhaustive-deps
 	const data = useAppSelector((state) => state.product.products) || [];
+	console.log('isloaddding', isLoading);
```
Comment (Angemichel12) [inline src/pages/dashboard/seller/SellerProductsPage.tsx]: remove this

File: src/pages/dashboard/seller/Sales/SalesPage.tsx (hunk 8 above)
Comment (YvetteNyibuka) [reply src/pages/dashboard/seller/Sales/SalesPage.tsx]: Removed

File: src/pages/dashboard/seller/Sales/SalesPage.tsx (hunk 7 above)
Comment (YvetteNyibuka) [reply src/pages/dashboard/seller/Sales/SalesPage.tsx]: Done

File: src/pages/dashboard/seller/Sales/SingleSale.tsx (hunk 9 above)
Comment (YvetteNyibuka) [reply src/pages/dashboard/seller/Sales/SingleSale.tsx]: done

File: src/pages/dashboard/seller/SellerProductsPage.tsx (hunk 10 above)
Comment (YvetteNyibuka) [reply src/pages/dashboard/seller/SellerProductsPage.tsx]: done

**Review by GarrixA — APPROVED** [review]

**Review by Angemichel12 — APPROVED** [review]

File: src/pages/dashboard/seller/SellerProductsPage.tsx (hunk 10 above)
Comment (YvetteNyibuka) [reply src/pages/dashboard/seller/SellerProductsPage.tsx]: removed

**Pull request merged**